testing/kuttl/e2e/exporter-password-change/00-assert.yaml

```diff
@@ -0,0 +1,20 @@
+apiVersion: kuttl.dev/v1beta1
+kind: TestAssert
+commands:
+# Check that all containers in the instance pod are ready
+- script: |
+    retry() { bash -ceu 'printf "$1\nSleeping...\n" && sleep 5' - "$@"; }
```

Comment:
"retry" here is basically sleep then exit with a message (maybe there is a better name for the function :thinking: ). If the pod isn't found, the script fails and Kuttl runs it again.

So the whole script will be run every time

Reply:
Ah, right, because what we discovered in testing was that the `commands` script in a TestAssert gets rerun from the beginning if any element fails.

Reply:
`exit_with_message()`
`wait_and_exit()` 

:thinking:

Reply:
How many times would it retry?

Reply:
the `retry` doesn't do anything special - it just waits and expects an exits 1. Kuttl will loop through the script until it exits 0 or hits a timeout

Reply:
Yeah, I guess this is a feature of TestAssert steps that has an unexpected side-effect when using `commands`:

* TestAsserts wait until their conditions are true -- so if you have a TestAssert looking for a pod with particular labels, KUTTL will keep checking until it finds that pod or until it times out.
* so if a `commands` section in a TestAssert fails, it just gets restarted.

If I put it that way, I get it, but I'm still surprised by that behavior the first few times I've seen it. Maybe worth a comment in the README?

Reply:
Yeah, i imagine a kuttl how to doc somewhere?

One thing of note - the script will loop until 1) it passes AND 2) the assert files are met. So even if a script exits 0, it will continue to run until the assert files (e.g., yaml asserts on a postgres cluster) also pass

Reply:
I didn't end up changing the name of the `retry` function - Will add to backlog